pkg/apis/crd/v1alpha2/types.go

```diff
@@ -543,3 +543,94 @@ type TrafficControlList struct {
 
 	Items []TrafficControl `json:"items"`
 }
+
+type BGPPolicyStatus string
+
+const (
+	BGPPolicyActive  = "Active"
+	BGPPolicyStandby = "Standby"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BGPPolicy defines BGP configuration applied to Nodes.
+type BGPPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BGPPolicySpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type BGPPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []BGPPolicy `json:"items"`
+}
+
+// BGPPolicySpec defines the specification for a BGP policy.
+type BGPPolicySpec struct {
+	// NodeSelector is utilized to select Nodes to which the policy is applied.
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector"`
+
+	// LocalASN is the AS number used by BGP protocol.
+	LocalASN int16 `json:"localASN"`
+
+	// BindAddresses is a list of IP addresses binding to BGP protocol.
+	BindAddresses []string `json:"bindAddresses,omitempty"`
+
+	// ListenPort is the port on which BGP protocol listen.
+	ListenPort *int16 `json:"listenPort,omitempty"`
```

Comment:
int32

int16 is not large enough anyway

Reply:
Will correct this.